### ISSUE TITLE
SERIAL_AUTO_CR inconsistently implemented

### ISSUE THREAD

**Issue by bennoleslie** [seL4/util_libs#48]
In some implementation (rockpro64) the `CR` is sent after the `LF`, while in others (such as imx) it is sent prior to the `LF`.

I *think* it should be sent prior (as in, `imx` is correct). Is there a reason the `rockpro64` is implemented differently?

**Comment by axel-h** [comment]
yes this is a cop&paste bug that keeps spreading. I've already upstreamed a few patches that fix this for the platform I have access to.
